batch-oai-harvest/src/main/java/data/unit/reader/OaiIdentifiersReader.java

```diff
@@ -0,0 +1,35 @@
+package data.unit.reader;
+
+import data.entity.ExecutionRecordExternalIdentifier;
+import data.repositories.ExecutionRecordExternalIdentifierRepository;
+import org.springframework.batch.core.configuration.annotation.StepScope;
+import org.springframework.batch.item.data.RepositoryItemReader;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.List;
+
+@Component
+@StepScope
+public class OaiIdentifiersReader extends RepositoryItemReader<ExecutionRecordExternalIdentifier> {
+
+    @Value("#{stepExecution.jobExecution.jobInstance.id}")
+    private Long jobInstanceId;
+
+    private final ExecutionRecordExternalIdentifierRepository executionRecordExternalIdentifierRepository;
+
+    public OaiIdentifiersReader(ExecutionRecordExternalIdentifierRepository executionRecordExternalIdentifierRepository) {
+        this.executionRecordExternalIdentifierRepository = executionRecordExternalIdentifierRepository;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        this.setRepository(executionRecordExternalIdentifierRepository);
+        this.setSort(Collections.emptyMap());
+        this.setMethodName("findAllByExecutionRecordKey_ExecutionId");
+        this.setArguments(List.of(jobInstanceId+""));
+
+        super.afterPropertiesSet();
```

Comment:
That is a good point actually! Perhaps we should rethink how we can set the values here and in the DefaultRepositoryItemReader(which does not run the super method) so that we don't override it's super method, or invoke it there as well.

For the second part you are right, it is not necessary.